config/config.go

```diff
@@ -25,13 +26,16 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-func LoadFile(paths []string, expandEnvVars bool) (*Config, error) {
+func LoadFile(logger *slog.Logger, paths []string, expandEnvVars bool) (*Config, error) {
 	cfg := &Config{}
 	for _, p := range paths {
 		files, err := filepath.Glob(p)
 		if err != nil {
 			return nil, err
 		}
+		if len(files) == 0 && logger != nil {
```

Comment:
The logger should never be nil.

```suggestion
		if len(files) == 0 {
```

Reply:
Also, let's put the nil check at the top of the function and return an error if no logger is passed.

Reply:
I only put the nil check in for the benefit of the unit tests. If you are happy to *require* a non-nil logger, then we can drop that test.